docs/cli/ilab-local-config.md

```diff
@@ -0,0 +1,184 @@
+# InstructLab CLI Local Storage Proposal
+
+## Why is this needed?
+
+For tools to be useful to the user, they generally need to work on and know about data pertinent to the user. For example, `podman` and `docker` track image blobs, `kubectl` tracks cluster credentials, etc.
+
+The InstructLab CLI manages a number of settings today such as the teacher model, taxonomy location, inference model, etc. using a local file known as `config.yaml`.
+
+While this method allows ilab to avoid additional complexity of managing a dedicated directory, it’s limited by its ability to scale beyond constraining the user to run `ilab` in a single directory.
+Since the new ilab CLI will need to not only manage datasets, downloaded models, and any intermediary files, it'll also have to make it simple for the user to orchestrate these things.
+
+We presently store these paths directly in the config, however this
+isn't sustainable as data is frequently moved around unbeknownst to the program. This leaves the user with two choices: either update
+the config to correctly point to either a relative path in the format
+of `../../uncle-dir/dataset.jsonl`, or worse, have to specify an absolute path.
+
+
+## What's being proposed?
+
+We propose the following:
+
+1. `ilab` will have a dedicated config directory seated at `~/.config/instructlab` on Linux/MacOS systems, and the `%APPDATA%` equivalent if Windows support is added in the future.
```

Comment:
Is your intent to support XDG?:
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html#introduction

Not arguing that you should necessarily but if you intended to I would change the text to be seated at $XDG_DATA_HOME/instructlab (.local/share/instructlab)

Reply:
Yes that's what I was thinking of.

Reply:
I strongly recommend to use https://pypi.org/project/platformdirs/ . It has the correct defaults for all important platforms.

I also like to point out that the use of XDG directories does not work well with systemd services. @mairin just mentioned that she wants systemd service support for InstructLab. XDG directories are problematic, because they are designed for human end-users with a home directory. `systemd --system` need an entirely different directory structure.

Reply:
@tiran Thanks for pointing this out! I had initially suggested XDG thinking of the laptop-targeted experience. But if we want a systemd service we'll need to use system directories. Ideally we'd use XDG for the laptop-target user experience and system-wide directories for the big mama GPU hw experience.

Reply:
^ to this end, @tiran suggests `I would store the paths in config.yaml and have a option to either use XDG or a common base directory for all paths.`